addons/sale_pdf_quote_builder/wizards/sale_pdf_quote_builder_dynamic_fields_wizard.py

```diff
@@ -0,0 +1,134 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+import json
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import ValidationError
+
+from odoo.addons.sale_pdf_quote_builder.const import DEFAULT_FORM_FIELD_PATH_MAPPING
+
+
+class SalePDFQuoteBuilderDynamicFieldsWizard(models.TransientModel):
+    _name = 'sale.pdf.quote.builder.dynamic.fields.wizard'
+    _description = "Sale PDF Quote Builder Dynamic Fields Configurator Wizard"
+
+    current_form_fields = fields.Json()
+    dynamic_fields_wizard_line_ids = fields.One2many(
+        'sale.pdf.quote.builder.dynamic.fields.wizard.line', 'dynamic_fields_wizard_id'
+    )
+
+    @api.onchange('current_form_fields')
+    def _get_wizard_lines(self):
+        """ Add wizard lines containing existing mappings and form fields from the current document.
+        """
+        for wizard in self:
+            wizard.dynamic_fields_wizard_line_ids = [Command.clear()]
+            form_field_path_map = json.loads(self.env['ir.config_parameter'].sudo().get_param(
+                'sale_pdf_quote_builder.form_field_path_mapping', DEFAULT_FORM_FIELD_PATH_MAPPING
+            ))
+            current_form_fields = json.loads(
+                wizard.current_form_fields
+            ) if wizard.current_form_fields else {}
+
+            # Get the form fields not yet in the config parameters
+            lines_vals = []
+            for doc_type, form_fields in current_form_fields.items():
+                for form_field in form_fields:
+                    if form_field not in form_field_path_map[doc_type]:
+                        lines_vals.extend([
+                            {'document_type': doc_type, 'form_field': form_field, 'path': ''}
+                        ])
+            # Add existing values from the config parameter
+            lines_vals.extend([
+                {'document_type': doc_type, 'form_field': form_field, 'path': path}
+                for doc_type, mapping in form_field_path_map.items()
+                for form_field, path in mapping.items()
+            ])
+            # Create the wizard lines
+            wizard.dynamic_fields_wizard_line_ids = [Command.create(vals) for vals in lines_vals]
+
+    def validate_dynamic_fields_configuration(self):
+        """ Save this mapping configuration in the config parameters. """
+        for wizard in self:
```

Comment:
Yet it's usually done, by convention.

Reply:
For compute, search, inverse/... methods, not buttons, especially wizard ones

Reply:
Should we rewrite 75% of payment's wizards then? :thinking: 
From what I was told, even if not needed, it's better to always do it. And we should tag only methods that can't (not that aren't) be batch called as `self.ensure_one()`. This is for consistency's sake, to avoid problems of mindless c/c...
I don't mind taking it off, but being told every x PR to change the way to do it is bothersome. So better agree on something once and for all. =]

Reply:
75% of payment wizard?
I'm only speaking of non 'orm' methods, this is a method only called on button click, and there is no specific reason to support it for multiple records (and I felt like it would be clearer without the indentation).
In the payment wizards, there is only the refund wizard `action_refund`, which is a 'dumb' method.
Anyway, not worth such a long debate, forget it.